src/components/VerticalNavigation/VerticalNavigation.js

```diff
@@ -0,0 +1,379 @@
+        this.updateMobileMenu();
+        this.setState({ showMobileNav: true });
+      }
+    } else if (navCollapsed) {
+      this.expandMenu();
+    } else {
+      this.collapseMenu();
+    }
+    onMenuToggleClick && onMenuToggleClick();
+  }
+
+  collapseMenu() {
+    const { onCollapse } = this.props;
+    this.setState({
+      navCollapsed: true,
+      explicitCollapse: true
+    });
+    onCollapse && onCollapse();
+    // TODO only in uncontrolled mode / with a specific prop
+    getBodyContentElement().classList.add('collapsed-nav');
+  }
+
+  expandMenu() {
+    const { onExpand } = this.props;
+    this.setState({
+      navCollapsed: false,
+      explicitCollapse: false
+    });
+    onExpand && onExpand();
+    // TODO only in uncontrolled mode
+    getBodyContentElement().classList.remove('collapsed-nav');
+
+    // Dispatch a resize event when showing the expanding then menu to
+    // allow content to adjust to the menu sizing
+    // TODO FIXME -- do we need this, and wouldn't we want it after the re-render?
+    // this.windowRef.nativeWindow.dispatchEvent(new Event('resize'));
+  }
+
+  updateHoverState(hovering, primary, secondary, tertiary) {
+    // Since the hideTimer can be longer than the hoverTimer, it is possible for two items to be "hovering" at a time.
+    // We must guard against this race condition, or a long-running hideTimer can undo a hover triggered later.
+    // We keep a count and only remove the hover styles if there are really no more hovered items in that tier.
+    const doUpdate = (numHoveredKey, hoverStateKey) => {
+      const newNumHovered = this.state[numHoveredKey] + (hovering ? 1 : -1);
+      this.setState({ [numHoveredKey]: newNumHovered });
+      if (hovering || newNumHovered < 1) {
+        this.setState({ [hoverStateKey]: hovering });
+      }
+    };
+    if (primary) {
+      if (secondary) {
+        doUpdate('numHoveredSecondary', 'hoverTertiaryNav');
+      } else {
+        doUpdate('numHoveredPrimary', 'hoverSecondaryNav');
+      }
+    }
+  }
+
+  updateNavOnItemHover(primary, secondary, tertiary) {
+    const { onItemHover } = this.props;
+    this.updateHoverState(true, primary, secondary, tertiary);
+    if (onItemHover) {
+      onItemHover(primary, secondary, tertiary);
+    }
+  }
+
+  updateNavOnItemBlur(primary, secondary, tertiary) {
+    const { onItemBlur } = this.props;
+    // if (!this.primaryHover()) hoverSecondaryNav: false ? from ng, is it necessary?
+    this.updateHoverState(false, primary, secondary, tertiary);
+    if (onItemBlur) {
+      onItemBlur(primary, secondary, tertiary);
+    }
+  }
+
+  updateNavOnItemClick(primary, secondary, tertiary) {
+    const { inMobileState, onItemClick } = this.props;
+    const item = deepestOf(primary, secondary, tertiary);
+    if (inMobileState) {
+      if (item.subItems && item.subItems.length > 0) {
+        this.updateMobileMenu(primary, secondary, tertiary); // TODO figure out what this did in ng
```

Comment:
In mobile state, for primary items with secondary children, a click results in showing the secondary nav menu. Same for secondary items with tertiary children showing the tertiary nav menu.

Reply:
Yep, figured that out earlier and I'm still getting around to cleaning up these old comments.